timemachine/cpp/src/context.cu

```diff
@@ -49,19 +54,17 @@ bool is_barostat(std::shared_ptr<Mover> &mover) {
     return false;
 }
 
-void Context::_verify_box(cudaStream_t stream) {
+void Context::_verify_box(const double *box_buffer, cudaStream_t stream) {
     // If there are no nonbonded potentials, nothing to check.
     if (nonbonded_pots_.size() == 0) {
         return;
     }
-    std::vector<double> h_box(9);
-    gpuErrchk(cudaMemcpyAsync(&h_box[0], d_box_t_, 9 * sizeof(*d_box_t_), cudaMemcpyDeviceToHost, stream));
```

Comment:
Previously were doing a copy from device -> device AND device -> host. Now we just do the later.